src/streaming-ingestion/pipeline/streaming-ingestion-pipeline.ts

```diff
@@ -0,0 +1,269 @@
+import { Construct } from 'constructs';
+import { StreamingIngestionPipelineCfnProps } from './model';
+import { addCfnNagSuppressRules } from '../../common/cfn-nag';
+import { createLambdaRole } from '../../common/lambda';
+import { attachListTagsPolicyForFunction } from '../../common/lambda/tags';
+import { POWERTOOLS_ENVS } from '../../common/powertools';
+import { SolutionNodejsFunction } from '../../private/function';
+
+export class StreamingIngestionPipeline extends Construct {
+  readonly streamingIngestionPipelineRole: IRole;
+  constructor(scope: Construct, id: string, props: StreamingIngestionPipelineCfnProps) {
+    super(scope, id);
+
+    this.streamingIngestionPipelineRole = new Role(this, 'StreamingIngestionPipelineRole', {
+      assumedBy: new CompositePrincipal(new AccountPrincipal(Aws.ACCOUNT_ID), new ServicePrincipal('kinesisanalytics.amazonaws.com')),
+      inlinePolicies: {
+        'streaming-ingestion-pipeline': new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              actions: [
+                'kinesisanalytics:CreateApplication',
+                'kinesisanalytics:DescribeApplication',
+                'kinesisanalytics:UpdateApplication',
+                'kinesisanalytics:StartApplication',
+                'kinesisanalytics:StopApplication',
+                'kinesisanalytics:DeleteApplication',
+              ],
+              resources: ['*'],
+            }),
+            new PolicyStatement({
+              actions: [
+                'kinesis:CreateStream',
+                'kinesis:StartStreamEncryption',
+                'kinesis:DeleteStream',
+                'kinesis:DescribeStream',
+                'kinesis:GetShardIterator',
+                'kinesis:GetRecords',
+                'kinesis:DescribeStreamSummary',
+              ],
+              resources: [
+                Arn.format({
+                  service: 'kinesis',
+                  resource: 'stream',
+                  resourceName: '*',
+                  arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
+                }, Stack.of(this)),
+              ],
+            }),
+            new PolicyStatement({
+              actions: [
+                'kinesis:PutRecord',
+                'kinesis:PutRecords',
+              ],
+              resources: [
+                Arn.format({
+                  service: 'kinesis',
+                  resource: 'stream',
+                  resourceName: '*',
+                  arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
+                }, Stack.of(this)),
+              ],
+            }),
+            new PolicyStatement({
+              actions: [
+                'kinesis:ListShards',
+                'kinesis:ListStreams',
+              ],
+              resources: [
+                '*',
+              ],
+            }),
+            new PolicyStatement({
+              actions: [
+                'iam:PassRole',
```

Comment:
why need it for kinesisanalytics?

Reply:
It will get Error when createKinesisAnalyticsApplication if not authorized to perform `iam:PassRole`. 
<img width="2163" alt="image" src="https://github.com/awslabs/clickstream-analytics-on-aws/assets/79438022/d5c9decd-f585-4c61-a20e-c91ab27efa97">

Reply:
Got it. But we need to specify the specific resource that is managed by ourselves.